src/Symfony/Component/Validator/Constraints/Base64Validator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class Base64Validator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Base64) {
+            throw new UnexpectedTypeException($constraint, Base64::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if ($constraint->allowDataUri) {
+            $value = preg_replace('/^data:.+;base64,/', '', $value);
```

Comment:
```suggestion
            $value = preg_replace('/^data:.++;base64,/', '', $value);
```

Maybe?

Reply:
I admit I don't know this quantifier and had hard times understand how this works :smile: Unfortunately, it makes  the tests fail with `data:image/png;base64,dGVzdA==` :confused: